src/Obsidian/Cache.ts

```diff
@@ -115,6 +116,17 @@ export function getTasksFromFileContent2(
                 sectionIndex++;
                 tasks.push(task);
             }
+        } else {
+            // We do not want root listItems to keep same behavior as DataView plugin
+            if (listItem.parent < 0) {
+                continue;
+            }
```

Comment:
### On the comment:

I don't understand this comment - it may just be an English thing...
I think it's saying that in order to be compatible with the Dataview plugin's handling of list items and tasks, we discard any root list items.

### On the `if` condition...

I don't understand how this works if the parent falls on line number zero, as negative zero is still zero...
Could you add a clarifying comment please about that behaviour, if indeed the behaviour is as intended...

Reply:
added comments for both

Reply:
@stasmarkin I think what @claremacrae meant (if I may) is that this looks like a local solution when the number is below zero and does not take into account hierarchies that consist of a `Task` at the end of several `ListItem`s. I think https://github.com/stasmarkin/obsidian-tasks/pull/3 fixes that issue, please merge in your fork and forward here.

@claremacrae do you agree?

Reply:
It's not quite that specific - it's that I don't understand how the code works for any contents of a list that starts on line 0... And suspect that there is a bug lurking there....

Reply:
In my understanding there is a bug indeed: in a hierarchy that contains a task under several list items or starts with a list item non at the beginning of the file, the task is not the root with null parent but has a list item parent.

my PR that I linked in my previous message fixes this behaviour.

Reply:
Well, talking about https://github.com/stasmarkin/obsidian-tasks/pull/3 , I don't have any strong preference here. 
In my taste, comparing listItem.parent with 0 is obvious and straight forward. If parent is below zero, then listItem is root. But checking if there is anything in line2ListItem for lineNumber is a bit more complex to understand. It rises so many questions like "what if listItem.parent if greater then 0, but we still didn't find anything in line2ListItem?", "could someone delete an entry in line2ListItem before?" and so on

Reply:
Ok, I could share this opinion. What about the tests? Please see the change in behaviour there.

Reply:
Thank you both very much for the comments and helpful discussion...

I generally find that if there is still more to discuss after 2 or 3 rounds of discussion in the comments here, it's a lot more effective to pick it up in a video call...

Since the currently-merged code is already clearly an improvement, I propose to:

- merge this as-is, once I have got the next Tasks release out..
- leave you both to decide what to do about the tests and/or behaviour next time you meet.

Thanks again! 🎉